pkgs/development/python-modules/mujoco/default.nix

```
@@ -0,0 +1,76 @@
+{ lib
+, pkgs
+, python3Packages
+, fetchPypi
+, glfw
+, python3
+}:
+
+let mujoco = pkgs.mujoco;
+in python3Packages.buildPythonPackage rec {
+  pname = "mujoco";
+  version = "3.0.0";
+
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-2ImKDfm9hCORaH7iou/Lp979VSCugCM/Gqnb/CkTeOw=";
+  };
+
+  nativeBuildInputs = with python3Packages; [ cmake setuptools ];
+  dontUseCmakeConfigure = true;
+  buildInputs = [
+    mujoco
+
+    # non-numerical
+    python3Packages.pybind11
+    python3Packages.glfw
+    glfw
+  ];
+  propagatedBuildInputs = [
+    # explicit, so that system package is not chosen
+    python3Packages.glfw
+
+    python3Packages.numpy
+  ];
+
+  pythonImportsCheck = [ "${pname}" ];
+
+  env.MUJOCO_PATH = "${mujoco}";
+  env.MUJOCO_PLUGIN_PATH = "${mujoco}/lib";
+
+  preConfigure =
+    # Use system packages for pybind, glfw
+    ''
+      sed -i '185s/.*/  ON/' mujoco/CMakeLists.txt # pybind
+      sed -i '79s/.*/  ON/'  mujoco/simulate/cmake/SimulateDependencies.cmake # glfw
+    '' +
+    # Use non-system eigen3, lodepng, abseil: Remove mirror info and prefill
+    # dependency directory. $build from setuptools.
+    (let
+      # E.g. 3.11.2 -> "311"
+      pythonVersionMajorMinor = with lib.strings;
+        with lib.lists;
+        concatStrings
+        (reverseList (drop 1 (reverseList (splitString "." python3.version))));
```

```suggestion
      pythonVersionMajorMinor = with lib.versions; "${major pythonVersion}${minor pythonVersion}";
```
something like this ought to do the trick. check out [lib.versions](https://github.com/NixOS/nixpkgs/blob/master/lib/versions.nix) for other helpy helpers